Snakefile

```diff
@@ -1,102 +1,481 @@
 
-rule filter:
+rule curate:
     input:
-        seq = seq_file,
-        meta = meta_file,
-        exclude = exclude_file
+        metadata_raw = "data/metadata_raw.tsv",
+        annotations=config["curate"]["annotations"],
     output:
-        "results/filtered.vcf.gz"
+        metadata_curate = "data/metadata_curate.tsv"
+    log:
+        "logs/curate.txt"
+    benchmark:
+        "benchmarks/curate.txt"
+    params:
+        field_map=format_field_map(config["curate"]["field_map"]),
+        strain_regex=config["curate"]["strain_regex"],
+        strain_backup_fields=config["curate"]["strain_backup_fields"],
+        date_fields=config["curate"]["date_fields"],
+        expected_date_formats=config["curate"]["expected_date_formats"],
+        articles=config["curate"]["titlecase"]["articles"],
+        abbreviations=config["curate"]["titlecase"]["abbreviations"],
+        titlecase_fields=config["curate"]["titlecase"]["fields"],
+        annotations_id=config["curate"]["annotations_id"],
+        id_field=config["curate"]["output_id_field"],
     shell:
         """
-        augur filter --sequences {input.seq} \
-            --metadata {input.meta} \
-            --exclude {input.exclude} \
-            --output {output}
+        augur curate passthru \
+            --metadata {input.metadata_raw} \
+            | augur curate rename \
+                --field-map {params.field_map} \
+            | augur curate normalize-strings \
+            | augur curate transform-strain-name \
+                --strain-regex {params.strain_regex} \
+                --backup-fields {params.strain_backup_fields} \
+            | augur curate format-dates \
+                --date-fields {params.date_fields} \
+                --expected-date-formats {params.expected_date_formats} \
+                --failure-reporting silent \
+            | augur curate titlecase \
+                --titlecase-fields {params.titlecase_fields} \
+                --articles {params.articles} \
+                --abbreviations {params.abbreviations} \
+            | augur curate apply-record-annotations \
+                --annotations {input.annotations} \
+                --id-field {params.annotations_id} \
+                --output-metadata {output.metadata_curate} 2> {log}
+        """
+
+checkpoint filter_subsample:
+    input:
+        metadata_curate = "data/metadata_curate.tsv"
+    output:
+        metadata_subsample = "data/metadata_subsample.tsv"
+    log:
+        "logs/filter_subsample.txt",
+    benchmark:
+        "benchmarks/filter_subsample.txt"
+    shell:
+        """
+        augur filter \
+            --metadata {input.metadata_curate} \
+            --metadata-id-columns accession \
+            --query "(mbases > 180 & mbases < 1500 & (country != 'Uncalculated'))" \
+	        --group-by country year \
+	        --sequences-per-group 1 \
+	        --min-date 1990 \
+	        --output-metadata {output.metadata_subsample} 2> {log}
```

Comment:
The capturing of STDOUT/STDERR differs from rule to rule at the moment. For non-piped commands like this one I think we universally want to capture both via either `2>&1 | tee {log:q}` or `exec &> >(tee {log:q})`.

I've not made comments on each rule, but I'd suggest going through each one and checking.